### PR TITLE
chore(appium): experimental build via tsc

### LINKED ISSUE #17746: dev script is broken: too many workspaces
So basically if you run `npm run dev` what _should_ happen is that the `dev` script runs in every package.  

There are like 22 packages in there, and a dev env probably doesn't have that many cores available, so only _some non-deterministic subset_ of those run.  This is not an issue for the `build` script since those scripts do not stay running indefinitely.

Need to figure out how to shoehorn this into a single command.  IDK maybe `nx` has a solution for this, since `lerna` now uses it under the hood.

### PR DESCRIPTION
**Pull request by boneskull**
See #17807.

This shows that `appium` can be built via `tsc`.

Here's [a gist which shows the output difference](https://gist.github.com/boneskull/a0d3b3659a32830335ed7b9f87bfe31e)

### CODE REVIEW

File: config/tsconfig.base.json [hunk 1]
```diff
@@ -7,12 +7,12 @@
...
     "declarationMap": true,
     "emitDeclarationOnly": true,
-    "module": "es2015",
+    "module": "CommonJS",
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "strictNullChecks": true,
     "removeComments": false,
-    "target": "es2015",
+    "target": "ESNext",
```
Comment (jlipps) [inline config/tsconfig.base.json:15]: don't we want to make sure the target matches our lowest version of node?

Comment (boneskull) [reply config/tsconfig.base.json:15]: hmm, yeah, probably.  I'm not sure what that is.  I'll have to figure it out.

Comment (boneskull) [reply config/tsconfig.base.json]: OK, this is ES2020 for Node.js v14.17.0 as per [the kangax table](https://kangax.github.io/compat-table/es2016plus/)

**Comment by boneskull** [comment]
If merged, this will close #17746

**Comment by boneskull** [comment]
@jlipps @mykola-mokhnach @KazuCocoa 

This looks pretty good to me.  Should I expand this to compile the whole monorepo?

**Review by mykola-mokhnach — APPROVED** [review]

**Review by KazuCocoa — APPROVED** [review]
oh, looks cool to me, 🚀

**Review by jlipps — APPROVED** [review]
awesome! lgtm once CI is happy. and did you manually run scripts that aren't run in CI just to double check they still work?

File: .npmrc [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # Do not disable package-lock. Disabling package-lock also disables npm-shrinkwrap
 package-lock=true
 save-exact=true
-
+foreground-scripts=true
```
[the remark this answers is not in the document]
Comment (boneskull) [reply .npmrc:4]: it's new-ish. they added it because people were begging in `postinstall` scripts.

**Comment by boneskull** [comment]
yeah I don't think this will work but it's a nice thought

```bash
for $script in $(cat package.json | '.scripts | keys[]'); do 
  npm run $script
done`
```

**Comment by boneskull** [comment]
@jlipps btw the only reason CI is failing is because stuff that was silently failing no longer does so silently.

**Comment by boneskull** [comment]
@jlipps OK, so I removed the "build documentation" job from CI because it hasn't been working in quite some time and it was failing silently.  Historically, it was trying to run the `parse-yml-commands` script, but that script is looking for files that no longer exist (`commands-yml/`, `template.md`).

**Pull request merged**